xclim/indices/_threshold.py

```diff
@@ -306,6 +346,64 @@ def growing_degree_days(
     )
 
 
+@declare_units("", tas="[temperature]", thresh="[temperature]")
+def growing_season_end(
+    tas: xarray.DataArray,
+    thresh: str = "5.0 degC",
+    mid_date: str = "07-01",
+    window: int = 5,
+    freq: str = "YS",
+):
+    r"""First day of consistent inferior threshold temperature.
+
+    Returns the first day of period where a temperature is inferior to a threshold
+    over a given run of days.
+
+    Parameters
+    ----------
+    tas : xarray.DataArray
+      Mean daily temperature [℃] or [K].
+    thresh : str
+      Threshold temperature on which to base evaluation [℃] or [K]. Default '5.0 degC'.
+    mid_date : str
+      Date of the year after which to look for the end of the season. Should have the format '%m-%d'.
+    window : int
+      Minimum number of days with temperature below threshold needed for evaluation.
+    freq : str
+      Resampling frequency. Default: "YS".
+
+    Returns
+    -------
+    xarray.DataArray
+      Day of the year when temperature is inferior to a threshold over a given number of days for the first time.
+      If there is no such day, return np.nan.
+    """
+    thresh = utils.convert_units_to(thresh, tas)
+    mid_doy = datetime.datetime.strptime(mid_date, "%m-%d").timetuple().tm_yday
+
+    def compute_growing_season_end(yrdata):
```

Comment:
Can this logic be packaged into a function and reused ?

Reply:
Potentially. I need to verify that this logic is not now redundant with @aulemahal 's recent work optimizing dask array functions. If it seems valuable still, I'll move it out and into a utility file.

Reply:
These (`compute_growing_season_end` and `compute_growing_season_length`) are wraps around the `run_length` methods to manage chunking, `mid_date` and sanity checks in the case of `season_length`.  If creating reusable functions, I suggest putting them in `run_length`, calling them something like :  `compute_run_(start|end|length)`.

Reply:
This seems doable. Working on this now.